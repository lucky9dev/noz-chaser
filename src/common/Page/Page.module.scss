.page {
  display: flex;
  flex-flow: column nowrap;
  gap: var(--spacing-2);
  height: 100%;
  padding: 0 var(--spacing-4);

  .header {
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
    color: var(--contrast);
  }
}
