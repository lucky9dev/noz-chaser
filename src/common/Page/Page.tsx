import React from 'react';
import { Back } from 'common';
import styles from './Page.module.scss';

interface PageProps {
  children: React.ReactNode;
  header: string;
}

const Page: React.FC<PageProps> = ({ children, header }) => {
  return (
    <section className={styles.page}>
      <div className={styles.header}>
        <h2>{header}</h2> <Back />
      </div>
      {children}
    </section>
  );
};

export default Page;
