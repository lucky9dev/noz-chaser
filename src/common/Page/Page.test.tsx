import { render } from '@testing-library/react';
import Page from './Page';
import '@testing-library/jest-dom';

describe('Page', () => {
  test('Page renders correctly', () => {
    const { container } = render(
      <Page header="Test">
        <div>Test</div>
      </Page>
    );
    expect(container).toMatchSnapshot();
  });
});
