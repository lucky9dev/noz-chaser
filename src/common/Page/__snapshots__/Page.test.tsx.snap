// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Page renders correctly 1`] = `
<div>
  <section
    class="page"
  >
    <div
      class="header"
    >
      <h2>
        Test
      </h2>
       
      <button
        class="ui button"
        style="height: 36px;"
        type="button"
      >
        <i
          aria-hidden="true"
          class="arrow left icon"
        />
        Back
      </button>
    </div>
    <div>
      Test
    </div>
  </section>
</div>
`;
